src/main/java/com/google/sps/data/Post.java

```diff
@@ -131,6 +145,51 @@ public void requestToPost(HttpServletRequest request, String collegeId) {
     Calendar nowTime = Calendar.getInstance(TimeZone.getTimeZone("America/Los_Angeles"));
     year = nowTime.get(Calendar.YEAR);
 
+    // Implements a try-catch statement in case the Servlet is accessed directly, instead of through Blob Upload URL.
+    try {
+      // If there is an image uploaded, set the Blob Key and Serving Url.
+      blobKey = getBlobKey(request, "foodImage");
+      imageServingUrl = getServingUrl(blobKey);
+      if (imageServingUrl == null) {
+        imageServingUrl = "no image";
+      }
+    }
+    catch (Exception e) {
+      // In the case the Blob Upload URL is not used, set the servingUrl and blobKey to default values.
+      System.out.println(e);
```

Comment:
With the new implementation, will this servlet exclusively be accessed through the Blob Upload URL, and this exception will only occur if somebody bypassed the frontend? Basically asking if this print statement would happen with regular usage of the app, or only in edge cases :)

Reply:
I recommend logging it instead of just printing!

Reply:
+1 to using a logger and Melanie's question.

Reply:
Great question! The POST will be executed exclusively through the Blob Upload URL. If not accessed through the URL, a `java.lang.IllegalStateException` occurs because the getBlobKey() function must be called from a blob upload callback request. There are a couple cases where the method will not be accessed with a callback request:
- A malicious user
- Testing

Good point about the logging, I added logging as described [here](https://cloud.google.com/appengine/docs/standard/java/how-requests-are-handled#Java_Logging)